src/fsharp/InfoReader.fs

```diff
@@ -700,7 +700,34 @@ let TryFindIntrinsicMethInfo infoReader m ad nm ty =
 /// Try to find a particular named property on a type. Only used to ensure that local 'let' definitions and property names
 /// are distinct, a somewhat adhoc check in tc.fs.
 let TryFindPropInfo infoReader m ad nm ty = 
-    GetIntrinsicPropInfosOfType infoReader (Some nm) ad AllowMultiIntfInstantiations.Yes IgnoreOverrides m ty 
+    GetIntrinsicPropInfosOfType infoReader (Some nm) ad AllowMultiIntfInstantiations.Yes IgnoreOverrides m ty
+    
+/// Try to find the method that overrides the given method from the given type.
+/// The given type must be an IL type.
+/// The given method must be an IL method.
+let TryFindILIntrinisicOverrideByMethInfo (infoReader: InfoReader) m ad ty (overrides: MethInfo) =
+    let g = infoReader.g
+
+    match tryAppTy g ty, overrides with
+    | ValueSome (tcref, _), ILMeth (_, ilMethInfo, _) when tcref.IsILTycon ->
+        let ilMeth = ilMethInfo.ILMethodRef
+
+        tcref.ILTyconRawMetadata.MethodImpls.AsList
+        |> List.tryFind (fun ilMethImpl -> 
+            let overridesRef = ilMethImpl.Overrides.MethodRef
+            overridesRef.Name = ilMeth.Name && overridesRef.ArgCount = ilMeth.ArgCount && 
+            overridesRef.GenericArity = ilMeth.GenericArity
```

Comment:
Could this be moved to the previous line, and/or have that line split by each check? It took a bit too much squinting to see that this was a part of the bigger check.

Reply:
This could look cleaner. To be honest, I did my best here to make it clean. You really don't want to know what it looked like before...